src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java

```diff
@@ -40,20 +40,16 @@
 import com.puppycrawl.tools.checkstyle.api.DetailAST;
 import com.puppycrawl.tools.checkstyle.api.FileContents;
 import com.puppycrawl.tools.checkstyle.api.FullIdent;
-import com.puppycrawl.tools.checkstyle.api.Scope;
 import com.puppycrawl.tools.checkstyle.api.TextBlock;
 import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.checks.naming.AccessModifier;
```

Comment:
> Strange that we do not have ImportControl on this.....

Checks are allowed to be called from everywhere, including from other checks.
https://github.com/checkstyle/checkstyle/blob/master/config/import-control.xml#L21

> As we started process of refactoring/deprecation of Scope, should we put @Deprecated on it ?

Yes. Shouldn't deprecation be its own issue as breaking compatibility so it stands out?

Reply:
> https://github.com/checkstyle/checkstyle/blob/master/config/import-control.xml#L21

Some other questionable violations when removed:
````
[checkstyle] [ERROR] \src\main\java\com\puppycrawl\tools\checkstyle\checks\annotation\MissingDeprecatedCheck.java:27:1: Disallowed import - com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck. [ImportControlMain]
[checkstyle] [ERROR] \src\main\java\com\puppycrawl\tools\checkstyle\checks\annotation\MissingDeprecatedCheck.java:28:1: Disallowed import - com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocTagInfo. [ImportControlMain]
[checkstyle] [ERROR] \src\main\java\com\puppycrawl\tools\checkstyle\checks\annotation\MissingOverrideCheck.java:30:1: Disallowed import - com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocTagInfo. [ImportControlMain]
[checkstyle] [ERROR] \src\main\java\com\puppycrawl\tools\checkstyle\checks\imports\UnusedImportsCheck.java:36:1: Disallowed import - com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocTag. [ImportControlMain]
````

Reply:
> Shouldn't deprecation be its own issue as breaking compatibility so it stands out?

I think we can put deprecation annotation at any commit. I pretty sure nobody will react till be remove this class, but it might be helpful to users who try to create new Check, this will warn him that better to use something else. 
up to you .... as you move AsseccModifier out, we should deprecate scope with link to AccessModifier.

Reply:
After one more thought ..... It is better to make it by separate issue .... to have a space for us explain users why this is required, why scope is problematic and .... probably let users have some feedback if they find this issue by google.

Reply:
we can improve it later on.
I recall this for now to let this PR be merged sooner.